### PR TITLE
Directional couplers

### PR DESCRIPTION
**Pull request by AKiselevLXT**
Implemented directional couplers with a compact model

## Summary by Sourcery

Implement a directional coupler with a compact model, including new cross-section definitions and simulation data documentation.

New Features:
- Introduce a directional coupler component with a compact model for photonic circuits.

Enhancements:
- Add new cross-section definitions for waveguides with widths of 0.7, 0.77, and 0.8 micrometers.

Documentation:
- Include a JSON file documenting the FTDT simulation results of the directional coupler for transmission at 1550 nm.

### CODE REVIEW

File: lnoi400/tech.py [hunk 1]
```diff
@@ -120,6 +120,51 @@ def get_layer_stack() -> LayerStack:
 # Cross-sections
 ############################
 
+xs_rwg700 = partial(
```
[the remark this answers is not in the document]
Comment (abeccari) [reply lnoi400/tech.py:133]: Please remove these cross-section definitions and add them directly in the dc cell with a function to tune the width. We want to expose to designers only commonly-used cross-sections in tech.py

File: lnoi400/models.py
Comment (abeccari) [inline lnoi400/models.py:1]: Same idea, please rename the model to reflect the updated cell name.

File: lnoi400/spline.py
Comment (abeccari) [inline lnoi400/spline.py:1]: Remove the print statements and debug instructions from production code.

File: lnoi400/tech.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (abeccari) [reply lnoi400/tech.py:133]: Note that I changed the declaration of PDK cross sections following the new gdsfactory updates, using the xsection decorator. See the merge commit.

**Review by abeccari — CHANGES_REQUESTED** [review]
Add a unit test of the directional coupler layout & model.

File: lnoi400/tech.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (AKiselevLXT) [reply lnoi400/tech.py:133]: Done

File: lnoi400/cells.py [hunk 2]
```diff
@@ -807,5 +811,112 @@ def chip_frame(
...
+        sbend_length: length of the s-bend part.
+        central_straight_length: length of the coupling region.
+        wg_sep: Distance between two waveguides in the coupling region.
+        cross_section_io: cross section width of the i/o (must be in tech.py).
+        cross_section_coupling: cross section width of the coupling section (must be in tech.py).
+    """
+
+    dc = gf.Component()
+
+    # top right branch
```
[the remark this answers is not in the document]
Comment (AKiselevLXT) [reply lnoi400/cells.py:1049]: Done! "directional_coupler_balanced" and "bend_S_spline_varying_width"

**Comment by abeccari** [comment]
@AKiselevLXT I implemented the changes to the cell arguments we discussed. The tests are also fixed. Please update the models notebook and we should be good to go!

File: docs/notebooks/models.ipynb
Comment (abeccari) [inline docs/notebooks/models.ipynb:1]: Please add markdown guiding the user through the new section of the models documentation.

**Review by abeccari — CHANGES_REQUESTED** [review]

**Review by abeccari — APPROVED** [review]

**Pull request merged**